mautic/mautic
# Fix #7319: GET /api/contacts return total as string

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | yes/no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #7319


#### Description:
Problem:

Endpoint `/api/contacts` return total value as `string` instead of as `int`.

##### Steps to reproduce

Set your `AUTH=user:password` and `MAUTIC_URL`, just to avoid exposing private config here. I added a `search` parameter just to return an empty response (it is not relevant for the bug).

```
curl -u $AUTH $MAUTIC_URL/api/contacts?search=do-not-exist
{
    "total": "0",
    "contacts": []
}
```

Other endpoints return as expected:

```
curl -u $AUTH $MAUTIC_URL/api/contacts?search=do-not-exist
{
    "total": 0,
    "lists": []
}
```

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test as above - you should see the total returned as an int rather than a string
